amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonschema/semanticjsonschema/transform/SchemaTransformer.scala

```diff
@@ -1,22 +1,95 @@
 package amf.shapes.internal.spec.jsonschema.semanticjsonschema.transform
 
+import amf.aml.client.scala.model.document.Vocabulary
+import amf.aml.client.scala.model.domain.{NodeMapping, PropertyMapping, UnionNodeMapping}
 import amf.aml.internal.metamodel.domain.NodeMappableModel
 import amf.core.client.scala.errorhandling.AMFErrorHandler
 import amf.core.client.scala.model.domain.DomainElement
-import amf.shapes.client.scala.model.domain.AnyShape
+import amf.shapes.client.scala.model.domain.{AnyShape, CuriePrefix, SemanticContext}
+import amf.shapes.internal.spec.jsonschema.semanticjsonschema.transform.SchemaTransformerOptions.DEFAULT
+import amf.shapes.internal.spec.jsonschema.semanticjsonschema.transform.SemanticOps.{expandIri, findPrefix}
 
 import scala.collection.mutable
 
 case class TransformationResult(encoded: DomainElement,
                                 declared: Seq[DomainElement],
-                                externals: mutable.Map[String, String])
+                                externals: Map[String, String],
+                                vocab: Option[Vocabulary] = None)
 
-case class SchemaTransformer(shape: AnyShape)(implicit errorHandler: AMFErrorHandler) {
+object SchemaTransformerOptions {
+  val DEFAULT = SchemaTransformerOptions("policy_vocabulary",
+                                         "https://anypoint.mulesoft.com/policy#",
```

Comment:
We should put something generic here as default. AMF is open source.
Some uri of aml with "semantic" instead of "policy" (policy is something specific of this use case)